### ISSUE TITLE
Blackman Harris FFT

### ISSUE THREAD

**Issue by jbienz** [hallidev/UnityWasapiAudio#7]
Hey @hallidev. Any chance you could help me add Blackman Harris FFT?

https://www.recordingblogs.com/wiki/blackman-harris-window

I did try to add it to `SpectrumBase` in `CalculateSpectrumPoints` but I couldn't get it to work right. The reason I'm asking for this is a visualization library in the asset store uses it and I like the results on the spectrum analyzer better. Unity has native support for this FFT in `AudioListener.GetSpectrumData`, but of course `AudioListener` doesn't support loopback.

Thank you sincerely for this amazing sample. Very very impressive.

**Comment by jbienz** [comment]
Never mind. I realized that CSCore added support for Window Functions back in 2018, they just never released a binary after they added the feature. I had to build CSCore from source and then I was able to use window functions.

**Comment by hallidev** [comment]
If you don't mind me asking, what exact changes did you have to make? I'm always happy to improve the results since every now and then I come back to this to toy with it. If you're interested, I'd be happy to take a PR with your changes

**Comment by jbienz** [comment]
Sure @hallidev, happy to contribute back. You're very generous for giving this code away.

I could submit a pull request, but honestly the changes to your code were quite minimal once I had CSCore updated. CSCore was really the challenging part since they haven't released a new binary in 3 years. I never could get the project to fully compile, but I got it building far enough to dump out a functional DLL.

Once you've updated the DLL, adding windowing support to your library is easy. Just go into [WasapiAudio.cs](https://github.com/hallidev/UnityWasapiAudio/blob/master/Assets/WasapiAudio/Scripts/Core/WasapiAudio.cs), and right after this line:

`_basicSpectrumProvider = new BasicSpectrumProvider(_soundInSource.WaveFormat.Channels, _soundInSource.WaveFormat.SampleRate, CFftSize);`

add this line:

`_basicSpectrumProvider.WindowFunction = WindowFunctions.HammingPeriodic;`

Now, that's not Blackman Harris, it's Hamming Periodic. But I think it's already significantly improved, and if I wanted to add Blackman Harris it's basically just adding one anonymous function.

Thank you again for this project. I learned so much!
